### ISSUE TITLE
Error compiling with debug flags

### ISSUE THREAD

**Issue by svigerske** [coin-or/Ipopt#137]
Issue created by migration from Trac.

Original creator: rln

Original creation time: 2010-10-19 21:27:00

Assignee: ipopt-team

Version: 3.8

CC:  rln@cmu.edu

When I configure using the debug flags --enable-debug --with-ipopt-verbosity=5 I get the errors shown below. The first one seems to be an error in spelling. In line 764 of 'src/Common/IpCachedResults.hpp' it says chached_results_, and it should be cached_results_. The second error I'm unsure on how to fix that, but its in the same file, line 765: the variable iter is not defined in this scope. 

Please see below for more details.

Thanks, 
  Rodrigo

```
g++ -DHAVE_CONFIG_H -I. -I../../../../../Ipopt/src/LinAlg/TMatrices -I../../../inc -I../../../../../Ipopt/src/LinAlg/TMatrices/../../Common -I../../../../../Ipopt/src/LinAlg/TMatrices/.. -g -pipe -pedantic-errors -Wimplicit -Wparentheses -Wreturn-type -Wcast-qual -Wall -Wpointer-arith -Wwrite-strings -Wconversion -Wno-unknown-pragmas -DCHECK_SMARTPTR -MT IpSymTMatrix.lo -MD -MP -MF .deps/IpSymTMatrix.Tpo -c ../../../../../Ipopt/src/LinAlg/TMatrices/IpSymTMatrix.cpp -o IpSymTMatrix.o
In file included from ../../../../../Ipopt/src/LinAlg/TMatrices/../IpVector.hpp:14,
                 from ../../../../../Ipopt/src/LinAlg/TMatrices/../IpMatrix.hpp:12,
                 from ../../../../../Ipopt/src/LinAlg/TMatrices/../IpSymMatrix.hpp:13,
                 from ../../../../../Ipopt/src/LinAlg/TMatrices/IpSymTMatrix.hpp:13,
                 from ../../../../../Ipopt/src/LinAlg/TMatrices/IpSymTMatrix.cpp:9:
../../../../../Ipopt/src/LinAlg/TMatrices/../../Common/IpCachedResults.hpp: In member function 'void Ipopt::CachedResults<T>::DebugPrintCachedResults() const':
../../../../../Ipopt/src/LinAlg/TMatrices/../../Common/IpCachedResults.hpp:764: error: 'chached_results_' was not declared in this scope
../../../../../Ipopt/src/LinAlg/TMatrices/../../Common/IpCachedResults.hpp:765: error: 'iter' was not declared in this scope
In file included from ../../../../../Ipopt/src/LinAlg/TMatrices/../IpVector.hpp:14,
                 from ../../../../../Ipopt/src/LinAlg/TMatrices/../IpMatrix.hpp:12,
                 from ../../../../../Ipopt/src/LinAlg/TMatrices/IpGenTMatrix.hpp:13,
                 from ../../../../../Ipopt/src/LinAlg/TMatrices/IpGenTMatrix.cpp:9:
../../../../../Ipopt/src/LinAlg/TMatrices/../../Common/IpCachedResults.hpp: In member function 'void Ipopt::CachedResults<T>::DebugPrintCachedResults() const':
../../../../../Ipopt/src/LinAlg/TMatrices/../../Common/IpCachedResults.hpp:764: error: 'chached_results_' was not declared in this scope
../../../../../Ipopt/src/LinAlg/TMatrices/../../Common/IpCachedResults.hpp:765: error: 'iter' was not declared in this scope
In file included from ../../../../../Ipopt/src/LinAlg/TMatrices/../IpVector.hpp:14,
                 from ../../../../../Ipopt/src/LinAlg/TMatrices/../IpMatrix.hpp:12,
                 from ../../../../../Ipopt/src/LinAlg/TMatrices/IpGenTMatrix.hpp:13,
                 from ../../../../../Ipopt/src/LinAlg/TMatrices/IpTripletHelper.cpp:11:
../../../../../Ipopt/src/LinAlg/TMatrices/../../Common/IpCachedResults.hpp: In member function 'void Ipopt::CachedResults<T>::DebugPrintCachedResults() const':
../../../../../Ipopt/src/LinAlg/TMatrices/../../Common/IpCachedResults.hpp:764: error: 'chached_results_' was not declared in this scope
../../../../../Ipopt/src/LinAlg/TMatrices/../../Common/IpCachedResults.hpp:765: error: 'iter' was not declared in this scope
```

**Comment by svigerske** [comment]
Comment by @andrea5w created at 2011-06-15 16:33:27

fixed now in trunk revision ba38bbc9f1

**Comment by svigerske** [comment]
Comment by @andrea5w created at 2011-06-15 16:33:27

Resolution: fixed
